custom_from.php

```diff
@@ -134,10 +260,97 @@ public function message_compose($params)
                         $score = 1;
                     }
                 }
-            }
+
+                if ($score > 2 && !$should_override_identity) {
+                    break;
+                }
+
+                /** Remove autocomplete by JS frontend after page loaded */
+                $rcmail->output->set_env('identities', array());
+
+                break;
+            case self::DEFAULT_IDENTITY_OPTION:
+                $default_identity = $rcmail->user->get_identity();
+                $field_attrib = ['name' => '_from'];
+
+                foreach ($attrib as $attr => $value) {
+                    if (in_array($attr, ['id', 'class', 'style', 'size', 'tabindex'])) {
+                        $field_attrib[$attr] = $value;
+                    }
+                }
+
+                $field_attrib['onchange'] = rcmail_output::JS_OBJECT_NAME . ".change_identity(this)";
+                $select_from = new html_select($field_attrib);
+
+                $select_from->add(format_email_recipient($default_identity['email'], $default_identity['name']), $default_identity['identity_id']);
+
+                $attrib['content'] = $select_from->show($default_identity['identity_id']);;
         }
 
         self::set_state($compose_id, $address);
+
+        return $attrib;
+    }
+
+    /**
+     * Override headers fields from identity (bcc, reply-to), by plugin settings
+     */
+    public function compose_additional_headers($compose_id, $msg_uid, $reply_from, $attrib)
+    {
+        if (!self::get_state($compose_id) && $reply_from !== self::RECEIVING_EMAIL_WITH_DEFAULT_IDENTITY_OPTION) {
+            return $attrib;
+        }
+
+        $part = strtolower($attrib['part']);
+        $value = '';
+
+        switch ($reply_from) {
+            case self::RECEIVING_EMAIL_WITH_DEFAULT_IDENTITY_OPTION:
+                $default_identity = rcmail::get_instance()->user->get_identity();
+                $value = $default_identity[$part] ?? '';
+                $field_attrib = array();
+
+                if (!$value && $part == 'replyto') {
+                    $value = $default_identity['reply-to'] ?? '';
+                }
+            case self::RECEIVING_EMAIL_OPTION:
```

Comment:
Is the fallback from previous case to this one intended? If yes and given there are only 2 supported cases on this switch, maybe using simple conditions could be more readable.